sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java

```diff
@@ -221,24 +222,25 @@ public String getConfig(String key, String group) {
     @Override
     public Map<String, List<String>> getConfigList(String key, String group, boolean exactMatchFlag) {
         final KieResponse kieResponse;
-        String covertGroup = group.replace(KieConstants.SEPARATOR, KieConstants.CONNECTOR);
+        String convertedGroup = group.replace(KieConstants.SEPARATOR, KieConstants.CONNECTOR);
         if (exactMatchFlag) {
-            kieResponse = getKieResponse(key, covertGroup, exactMatchFlag);
+            kieResponse = getKieResponse(key, convertedGroup, exactMatchFlag);
         } else {
             kieResponse = getKieResponse(key, null, exactMatchFlag);
         }
         if (kieResponse == null || kieResponse.getData() == null) {
             return Collections.emptyMap();
         }
         Map<String, List<String>> result = new HashMap<>();
+        Pattern pattern = Pattern.compile(convertedGroup);
```

Comment:
use a map to cache these Pattern for better query efficiency

Reply:
done